Tests/LocalQueueServerRunnerTests/QueueServerConfigurationTests.swift

```diff
@@ -93,4 +93,22 @@ final class QueueServerConfigurationTests: XCTestCase {
             ]
         )
     }
+    
+    func test___deployment_destination_with_key_parsing() throws {
+        let data = Data("""
+            {
+                "host": "host",
+                "port": 1,
+                "username": "username",
+                "key": "key",
+                "remoteDeploymentPath": "/remote/deployment/path"
```

Comment:
Again, this test does not cover all permutations. It is better to come up with a more complex schema that will limit possible permutations to only two, e.g.

```json
{
    "authentication": {
        "type": "plain",
        "username": "user",
        "password": "pass"
    }
}
```

or

```json
{
    "authentication": {
        "type": "key",
        "key": "KEY_STRING"  // or is it /path/to/key_file?
    }
}
```